### PR TITLE
Update provider configuration docs

### PR DESCRIPTION
**Pull request by jrhouston**
### Description

This PR makes the docs for configuring the provider a bit more clear.

### CODE REVIEW

File: website/docs/index.html.markdown [hunk 1]
```diff
@@ -88,10 +93,27 @@ provider "helm" {
 }
 ```
 
-### In-cluster Configuration
+### In-cluster Config
+
+The provider is able to detect when it is running inside a cluster, so in this case you do not need to specify any attributes in the provider block.
```
Comment (redeux) [inline website/docs/index.html.markdown]: if you want to connect to the local kubernetes cluster.

File: website/docs/index.html.markdown [hunk 2]
```diff
@@ -70,7 +75,7 @@ provider "helm" {
 }
 ```
 
-### Statically defined credentials
+### Credentials config
 
 You can also **statically** define all the credentials:
```
Comment (redeux) [inline website/docs/index.html.markdown]: I wonder if the word **statically** could be misinterpreted here, when you're able to use variables.  Should it be **explicitly** or **discretely** instead?

Comment (jrhouston) [reply website/docs/index.html.markdown]: Yeah I was just thinking that – I think explicit makes more sense as being able to source these attributes from variables or outputs from other resources is not static at all

**Comment by redeux** [comment]
Thanks for doing this John.  

I wonder if this could be broken down as:

- File Config
- Explicit Config
  - defined provider auth attributes
- Implicit  Config
  - Environment variables
  - In-cluster

**Comment by jrhouston** [comment]
> Thanks for doing this John.
> 
> I wonder if this could be broken down as:
> 
> * File Config
> * Explicit Config
>   
>   * defined provider auth attributes
> * Implicit  Config
>   
>   * Environment variables
>   * In-cluster

Yeah totally – though I think that specifying the path to the config file is still explicit. I think noting the distinction between explicit configuration in the provider block and implicitly sourcing it from the environment is a good call.

**Comment by redeux** [comment]
> > Thanks for doing this John.
> > I wonder if this could be broken down as:
> > 
> > * File Config
> > * Explicit Config
> >   
> >   * defined provider auth attributes
> > * Implicit  Config
> >   
> >   * Environment variables
> >   * In-cluster
> 
> Yeah totally – though I think that specifying the path to the config file is still explicit. I think noting the distinction between explicit configuration in the provider block and implicitly sourcing it from the environment is a good call.

Agreed, file based config is a subset of explicit config.

File: website/docs/index.html.markdown [hunk 3]
```diff
@@ -39,25 +39,24 @@ resource "helm_release" "nginx_ingress" {
...
 
-1. [Using a kubeconfig file](#file-config)
-2. [Supplying credentials](#credentials-config)
-3. [Using the in-cluster config](#in-cluster-config)
-4. [Exec plugins](#exec-plugins)
 
 ### File config
 
-The easiest way is to supply a path to your kubeconfig file using the `config_path` attribute or using the `KUBE_CONFIG_PATH` environment variable.
+The easiest way is to supply a path to your kubeconfig file using the `config_path` attribute or using the `KUBE_CONFIG_PATH` environment variable. A kubeconfig file may have multiple contexts. If `config_context` is specified, the provider will use the `default` context.
```
Comment (aareet) [inline website/docs/index.html.markdown]: typo - If `config_context` is *not* specified

Comment (aareet) [reply website/docs/index.html.markdown]: I would also add a link to https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs#config_context

File: website/docs/index.html.markdown [hunk 4]
```diff
@@ -39,25 +39,24 @@ resource "helm_release" "nginx_ingress" {
 ## Requirements
 
 - You must have a Kubernetes cluster available. We recommend version 1.14.0 or higher.
```
Comment (aareet) [inline website/docs/index.html.markdown]: Should we change `recommend` to `support`?

File: website/docs/index.html.markdown [hunk 5]
```diff
@@ -100,7 +97,9 @@ provider "helm" {
 
 ### In-cluster Config
 
-The provider is able to detect when it is running inside a cluster, so in this case you do not need to specify any attributes in the provider block if you want to connect to the local kubernetes cluster. If you want to connect to a different cluster than the one terraform is running inside, configure the provider as above.
+The provider use the `KUBERNETES_SERVICE_HOST` and `KUBERNETES_SERVICE_PORT` environment variables to detect when it is running inside a cluster, so in this case you do not need to specify any attributes in the provider block if you want to connect to the local kubernetes cluster.
```
Comment (aareet) [inline website/docs/index.html.markdown]: The provider use -> uses

**Review by aareet — CHANGES_REQUESTED** [review]
LGTM, minor typos

**Review by redeux — APPROVED** [review]

**Pull request merged**